pkg/diagnostics/pod/auth.go

```diff
@@ -106,7 +106,10 @@ func (d PodCheckAuth) authenticateToRegistry(token string, r types.DiagnosticRes
 	if result, completed := dnsQueryWithTimeout(msg, resolvConf.Servers[0], 2); !completed {
 		r.Error("DP1006", nil, fmt.Sprintf("DNS resolution for registry address %s timed out; this could indicate problems with DNS resolution or networking", registryHostname))
 		return
-	} else if len(result.in.Answer) == 0 {
+	} else if result.err != nil {
+		r.Error("DP1016", nil, fmt.Sprintf("DNS resolution for registry address %s returned an error; container DNS is likely incorrect. The error was: %v", registryHostname, result.err))
```

Comment:
> I'm not sure it would improve things any, because you would replace the difficulty of checking for duplicate strings with the difficulty of ensuring you haven't reused something else's constant (via typo etc). Or maybe I'm not imagining the same implementation you are.

You normally end up with one big constant file and its very easy to inspect for duplicates (or even check via reflection), the names indicate what its for:

```go
DNSResolutionId = "DP1016"
```

The code is very readable and ids aren't re-used.  Pretty much every significant java project using resource bundles will have this approach for keys.  Eclipse even has tooling for chasing the ids through to the bundles.